documentation/cubic-pos-article.md

```diff
@@ -0,0 +1,156 @@
+
+
+# Epoched Proof of Stake
+
+The native staking token in Namada is identified as *NAM*. We call the smallest denomination of this token *namnam*, where 1 million *namnam* is equal to 1 *NAM*. An epoch defines a grouped set of consecutive blocks and a period of time that have some fixed configuration of the PoS system. For example, the set of validators who participate in the consensus protocol and their associated consensus voting powers can only change at the boundary of two epochs and are constant throughout the duration of one.
+
+Namada employs epoched staking for two main reasons:
+- Light clients become more efficient because there are fewer state changes (particularly validator set changes) to keep track of, resulting in quicker state verification
+- Eventually, Namada (and the larger Anoma ecosystem) will rely on the [Ferveo DKG protocol](https://eprint.iacr.org/2022/898) to encrypt transactions in the mempool. Ferveo must know the validator set for a near-future epoch in order to have sufficient time to compute the DKG in advance of the given epoch.
+
+Thus, changes to the state of the PoS system are enqueued to take effect at some epoch in the future, the lengths of which we refer to as *offsets* relative to the epoch in which the transaction is submitted. Most state changes, including validator set updates, take effect at the *pipeline offset* - 2 epochs into the future. The other important offset in the model is the *unbonding offset* of 21 epochs, primarily the length of time before an unbond becomes withdrawable.
+
+**Important parameters in the Namada PoS model (followed by their default values)**
+- Duration of a block: ~ 5 - 10 seconds
+- Duration of an epoch: ~ 1 day
+- Pipeline offset length: 2 epochs
+- Unbonding offset length: 21 epochs
+
+Below, we will review the rules around important PoS actions
+- Bonding tx submitted at epoch `n` initiates changes at epoch `n + pipeline_offset (n+2)`
+	- Bond becomes active, validator bonded stake and validator set is updated
+- Unbonding tx submitted at epoch `n`
+	- Updates validator set with decreased validator voting power at epoch `n + pipeline_offset (n+2)`
+	- Amount of the unbond is withdrawable at epoch `n + unbonding_offset + pipeline_offset (n+23)`
+- Redelegation tx submitted at epoch `n`
+	- Updates validator set with voting powers for old and new validators at epoch `n + pipeline_offset (n+2)`
+	- Only support redelegating 100% of the token amount of a bond
+	- A redelegated bond cannot be redelegated again until epoch `n + unbonding_offset + pipeline_offset (n+23)`
+	- Slashes are properly handled to ensure that only tokens that contributed to a validator's voting power at the time of infraction are affected
+
+**Validator Sets**
+
+Namada validators are classified into one of three sets:
```

Comment:
I think it's important to distinguish here that this is only in relation to consensus, due to the consensus validator set size limit. These are not relevant to e.g. governance

Reply:
thx, added the following sentence to the end of this section:
"Note that this construction of the validator sets is only important for the consensus mechanism; all validators can participate with their voting power in governance."